packages/models/src/lib/table.ts

```diff
@@ -0,0 +1,20 @@
+import { z } from 'zod';
+
+const headingsSchema = z.object(
+  {
+    key: z.string(),
+    label: z.string().optional(),
+  },
+  { description: 'Source file location' },
+);
+
+export type Headings = z.infer<typeof headingsSchema>;
+export const tableSchema = (description = 'Table information') =>
+  z.object(
+    {
+      headings: z.array(headingsSchema).optional(),
+      items: z.array(z.union([z.unknown(), z.record(z.string())])),
```

Comment:
I would love to have a generic schema... Any ideas?

Reply:
I think it should be possible, but complicated. Benefit would be to validate heading keys match item keys, and maybe give plugin author's (but not Lighthouse) a little extra autocomplete. I can give it a try tomorrow, but I think there's a good change it's more complexity than it's worth.